streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -349,6 +359,57 @@ public String queryableStoreName() {
         return toStream().selectKey(mapper);
     }
 
+    @Override
+    public KTable<K, V> suppress(final Suppressed<K, V> suppressed) {
+        final SuppressedImpl<K, V> suppressImpl = (SuppressedImpl<K, V>) suppressed;
+        final String name = builder.newProcessorName(SUPPRESS_NAME);
```

Comment:
I had a potentially kooky idea...

What if instead of bumping up the node index here, we just tack "SUPPRESS" on to the parent node name somehow?

This way, it would become fine to slap suppressions into topologies and restart without any other changes. This might be especially important if we intend to insert suppressions in the future for optimization reasons.

Thoughts?

Reply:
Could you elaborate a bit more on `just tack "SUPPRESS" on to the parent node name somehow?` what is the concrete proposal of the naming scheme?

Reply:
I should say: I don't want to add it to *this* PR (which is already large). If you think there's merit to this idea, I would tackle it in another PR.

Reply:
Because it does not help us, if somebody inserts a `filter()` for example. The overall renaming issue is, that inserting new operator results in re-indexing. This would be a "local" solution for `suppress()` only, but no global solution for all operators. Thus, my concern is, that we end up with different solutions for different operators for the same underlying issue. It's about consistency. Does this make sense?

Reply:
Yes, this is the downside I was concerned about. I won't make any change to this in this PR.

I've created a Jira to continue the discussion: https://issues.apache.org/jira/browse/KAFKA-7393